### PR TITLE
Add capability to select fields to dump

### PR DESCRIPTION
**Pull request by MaxThevenet**
Merge after https://github.com/ECP-WarpX/WarpX/pull/809. This should be ready for review, but I will keep the WIP label until PR #809 is merged.

Incremental change from PR #809: the user can choose the fields to plot. The technique is the same as was used for the plotfiles in the previous implementation.

Technical detail: functions to check if a string is in a vector of string have been moved from FieldsIO.cpp to WarpXUtils.H/.cpp, in namespace WarpXUtilStr.

### CODE REVIEW

File: Source/Utils/WarpXUtil.H [hunk 1]
```diff
@@ -141,4 +141,23 @@ void AlwaysAssert(bool is_expression_true, const std::string& msg);
...
 }
 
+namespace WarpXUtilStr
+{
+    /** Return true if elem is in vect, false otherwise
+     * @param[in] vect vector of strings, typically names
+     * @param[in] elem single string
+     * @return true if elem is in vect, false otherwise
+     */
+    bool is_in(std::vector<std::string> vect,
```
Comment (atmyers) [inline Source/Utils/WarpXUtil.H]: Probably better to pass these by const reference.

File: Source/Utils/WarpXUtil.H [hunk 2]
```diff
@@ -141,4 +141,23 @@ void AlwaysAssert(bool is_expression_true, const std::string& msg);
...
+               std::string elem);
+
+    /** Return true if any element in elems is in vect, false otherwise
+     * @param[in] vect vector of strings, typically names
+     * @param[in] elems vector of string
+     * @return true if any element in elems is in vect, false otherwise
+     */
+    bool is_in(std::vector<std::string> vect,
```
Comment (atmyers) [inline Source/Utils/WarpXUtil.H]: Also here.

File: Source/Utils/WarpXUtil.H (hunk 1 above)
Comment (MaxThevenet) [reply Source/Utils/WarpXUtil.H]: Done, thanks!

File: Source/Utils/WarpXUtil.H (hunk 2 above)
Comment (MaxThevenet) [reply Source/Utils/WarpXUtil.H]: Done, thanks!

**Pull request merged**